beefy-gadget/src/worker.rs

```diff
@@ -21,20 +21,22 @@ use std::{
 	sync::Arc,
 };
 
-use beefy_primitives::{
-	BeefyApi, Commitment, ConsensusLog, MmrRootHash, SignedCommitment, ValidatorSet, BEEFY_ENGINE_ID, KEY_TYPE,
-};
 use codec::{Codec, Decode, Encode};
 use futures::{future, FutureExt, StreamExt};
 use hex::ToHex;
 use log::{debug, error, info, trace, warn};
-use parking_lot::Mutex;
+use parking_lot::{Mutex, RwLock};
```

Comment:
What is actually the advantage of these versions over the `std::sync` ones?

Reply:
tldr: they are faster

There is some useful information here: https://github.com/Amanieu/parking_lot/blob/master/README.md https://webkit.org/blog/6161/locking-in-webkit/
It's likely that in the future the locks in the standard library will just be replaced by this implementation.